network-mux/src/Control/Concurrent/JobPool.hs

```diff
@@ -99,7 +101,10 @@ readGroupSize JobPool{jobsVar} group =
 
 collect :: MonadSTM m => JobPool group m a -> STM m a
 collect JobPool{completionQueue} = readTQueue completionQueue
-
+  -- GR-FIXME[C2]: better name?
+  --  - 'collect' gives the impression getting many and NOT blocking
+  --  - how about 'getCompletedJob'?
+  
```

Comment:
@dcoutts what about `getCompletedJob`, or maybe `waitForJob` (unlike `wait` from `async` it will not throw if the job fails).

Reply:
I used `waitForJob`, which conveys the blocking nature of it.